### PR TITLE
Send function information in manual instrumentation.

### PR DESCRIPTION
**Pull request by florian-kuebler**
This collects and sends information about the function
that contains the ORBIT_* macro, by collecting the
return address of the api functions.

Screenshots:
http://screen/ofPGwSrqCLPiG9L
http://screen/6kqNLh4Sy5v5CTQ

Bug: http://b/194764259
Test: Profile OrbitTest

### CODE REVIEW

File: src/OrbitGl/AsyncTrack.cpp [hunk 1]
```diff
@@ -62,15 +52,32 @@ AsyncTrack::AsyncTrack(CaptureViewElement* parent, TimeGraph* time_graph,
...
   }
-  std::string function_name = manual_inst_manager->GetString(event_id);
+  std::string label = manual_inst_manager->GetString(event_id);
+
+  std::string module_name = orbit_client_data::CaptureData::kUnknownFunctionOrModuleName;
+  std::string function_name =
+      capture_data_->GetFunctionNameByAddress(timer_info->address_in_function());
+
+  if (timer_info->address_in_function() != 0) {
+    const auto* function =
```
Comment (pierricgimmig) [inline src/OrbitGl/AsyncTrack.cpp:62]: auto ->  FunctionInfo

File: src/OrbitGl/AsyncTrack.cpp [hunk 2]
```diff
@@ -62,15 +52,32 @@ AsyncTrack::AsyncTrack(CaptureViewElement* parent, TimeGraph* time_graph,
...
+        capture_data_->FindFunctionByAddress(timer_info->address_in_function(), false);
+    if (function != nullptr) {
+      function_name = function->pretty_name();
+    }
+    const auto* module = capture_data_->FindModuleByAddress(timer_info->address_in_function());
```
Comment (pierricgimmig) [inline src/OrbitGl/AsyncTrack.cpp:67]: auto -> ModuleData

File: src/OrbitGl/ThreadTrack.cpp [hunk 3]
```diff
@@ -120,31 +120,41 @@ std::string ThreadTrack::GetBoxTooltip(const Batcher& batcher, PickingId id) con
...
+  if (func != nullptr) {
+    module_name = orbit_client_data::function_utils::GetLoadedModuleNameByPath(func->file_path());
+    function_name = label;
+  } else if (timer_info->address_in_function() != 0) {
+    function_name = capture_data_->GetFunctionNameByAddress(timer_info->address_in_function());
+
+    const auto* module = capture_data_->FindModuleByAddress(timer_info->address_in_function());
+    if (module != nullptr) {
+      module_name = module->name();
+    }
```
Comment (pierricgimmig) [inline src/OrbitGl/ThreadTrack.cpp:147]: nit. assign module_name before function_name to match above code ordering?

**Review by pierricgimmig — APPROVED** [review]
This is cool! Thanks for this Florian!

File: src/Api/Orbit.cpp [hunk 4]
```diff
@@ -35,13 +35,17 @@ void EnqueueApiEvent(Types... args) {
 extern "C" {
 
 void orbit_api_start(const char* name, orbit_api_color color) {
-  EnqueueApiEvent<orbit_api::ApiScopeStart>(name, color);
+  auto return_address =
+      absl::bit_cast<uint64_t>(__builtin_extract_return_addr(__builtin_return_address(0)));
```
Comment (ronaldfw) [inline src/Api/Orbit.cpp:39]: Add a comment about the meaning of "0" in the call.

[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/Api/Orbit.cpp:39]: Only for "level" > 0 this function is considered to be *unsafe*. For level == 0, it is trivial for the compiler to insert the return address.

[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/Api/Orbit.cpp:43]: Also, no framepointers are required for level 0.

Comment (florian-kuebler) [reply src/Api/Orbit.cpp:43]: Done.

File: src/OrbitGl/AsyncTrack.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/OrbitGl/AsyncTrack.cpp]: Done.

File: src/OrbitGl/AsyncTrack.cpp (hunk 2 above)
Comment (florian-kuebler) [reply src/OrbitGl/AsyncTrack.cpp]: Done.

File: src/OrbitGl/ThreadTrack.cpp (hunk 3 above)
Comment (florian-kuebler) [reply src/OrbitGl/ThreadTrack.cpp]: Sure.

**Pull request merged**